mastering/revisions.rst

```diff
@@ -0,0 +1,93 @@
+Package Revisions [EXPERIMENTAL]
+================================
+
+The goal of the `revisions feature` is to achieve package immutability, so nothing in a server is ever overwritten.
+
+.. note::
+
+    This is the first piece to achieve reproducibility: Recreate the exact dependency graph by using some
+    mechanism like a ``graph lock`` file. For example, if we store a ``graph lock`` file for the different releases
+    of our project, we can install the same dependencies just by using the graph lock.
+
+    **IMPORTANT:** The reproducibility is in the Conan roadmap and currently under development.
+
+
+How it works
+------------
+
+**In the client**
+
+- When a **recipe** is exported, Conan calculates a unique ID (revision). For every change,
+  a new recipe revision (RREV) will be calculated:
+
+   - If Conan detects SCM (Git or SVN) in the conanfile directory, the commit hash will be used as the RREV.
+   - If not SCM is detected, the checksum hash of the recipe manifest will be used as the RREV.
+
+.. note::
+
+   In future versions Conan will let the user choose to use the "recipe manifest" instead of the scm. For example,
+   when you use a mono-repo with N recipes, it is more convenient to calculate each
+   recipe revision based on its contents and not in the common github repository commit hash.
+   (https://github.com/conan-io/conan/issues/4413)
+
+
+- When a **package** is created (by running :ref:`conan create<conan_create>` or :ref:`conan export-pkg<conan_export-pkg>`)
+  a new package revision (PREV) will be calculated always using the hash of the package contents.
+  The packages and their revisions (PREVs) belongs to a concrete recipe revision (RREV).
+  The same package ID (for example for Linux/GCC5/Debug), can have multiple revisions (PREVs) that belong
+  to a concrete RREV.
+
+
+If a client request a reference like `lib/1.0@conan/stable`, Conan will retrieve automatically the latest revision.
+In the client cache there is only one revision installed at the same time.
+
+The revisions can be pinned when you write a reference (in the recipe requires, or in a reference in a
+conan install command…) but if you don’t specify a revision the server will retrieve the latest revision.
+
+You can specify the references in the following formats:
+
++---------------------------------------------+----------------------------------------------------------------+
+| Reference                                   | Meaning                                                        |
++=============================================+================================================================+
+| `lib/1.0@conan/stable`                      | Latest RREV for lib/1.0@conan/stable                           |
```

Comment:
We need kind of a "glosary" with all the Conan terminology: reference, recipe, package reference, revision... Not to be done in this PR